### PR TITLE
[ci-visibility] Intelligent test runner's unskippable tests for Javascript

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do? What is the motivation?
This PR adds instructions and examples on how to use the unskippable feature of CI Visibility's intelligent test runner. This will allow users to more safely use ITR.

### Merge instructions
- [x] Please merge after reviewing

### Additional notes
Preview: https://docs-staging.datadoghq.com/juan-fernandez/unskippable-tests-js/continuous_integration/intelligent_test_runner/setup/javascript/?tab=onpremisesciproviderdatadogagent#disabling-skipping-for-specific-tests

### CODE REVIEW

File: content/en/continuous_integration/intelligent_test_runner/setup/javascript.md [hunk 1]
```diff
@@ -19,33 +19,30 @@ further_reading:
...
+  * Run mocha with [`nyc`][1] to enable code coverage.
 * `cucumber-js>=7.0.0`
-  * From `dd-trace>=3.16.0` or `dd-trace>=2.29.0`.
-  * Run cucumber-js with [`nyc`][1] to enable [code coverage][5].
+  * From `dd-trace>=4.17.0` or `dd-trace>=3.38.0`.
+  * Run cucumber-js with [`nyc`][1] to enable code coverage.
 * `cypress>=6.7.0`
-  * From `dd-trace>=4.2.0`, `dd-trace>=3.23.0` or `dd-trace>=2.36.0`.
-  * Instrument your web application with [code coverage](#cypress)
+  * From `dd-trace>=4.17.0` or `dd-trace>=3.38.0`.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/setup/javascript.md:38]: it's not great that we keep increasing the minimum supported versions, but I'd say specifically unskippable tests is pretty important

File: content/en/continuous_integration/intelligent_test_runner/setup/javascript.md [hunk 2]
```diff
@@ -92,16 +89,126 @@ NODE_OPTIONS="-r dd-trace/ci/init" DD_ENV=ci DD_SERVICE=my-javascript-app DD_CIV
...
+{{< tabs >}}
+{{% tab "Jest/Mocha/Cypress" %}}
+You can use the following docblock at the top of your test file to mark a suite as unskippable. This prevents any of the tests defined in the test file from being skipped by Intelligent Test Runner. This is similar to jest's [`testEnvironmentOptions`][1].
+
+```javascript
+/**
+ * @datadog {"unskippable": true}
+ */
+
+// your tests continue normally
```
Comment (romainkomorndatadog) [inline content/en/continuous_integration/intelligent_test_runner/setup/javascript.md:126]: This just seems weird to me. The file continues normally? Tests continue to run normally?

Is this necessary?

**Review by romainkomorndatadog — APPROVED** [review]
I find it a bit weird that the "why prevent skipping tests?" and "example of unskippable tests" sandwiches the "how to mark tests unskippable" section, but other than that, makes sense.

**Review by anmarchenko — APPROVED** [review]

File: content/en/continuous_integration/intelligent_test_runner/setup/javascript.md (hunk 2 above)
Comment (juan-fernandez) [reply content/en/continuous_integration/intelligent_test_runner/setup/javascript.md:126]: I'll remove it. I was on the fence about it

**Comment by juan-fernandez** [comment]
> I find it a bit weird that the "why prevent skipping tests?" and "example of unskippable tests" sandwiches the "how to mark tests unskippable" section, but other than that, makes sense.

how would you order the sections?

**Review by hestonhoffman — APPROVED** [review]
LGTM! 
Re: the layout of this section. I don't see anything wrong with going from why -> how -> examples. I guess an argument could be made for moving the examples up into the "why" section, but I probably wouldn't. The "how" section is more important IMO.

**Pull request merged**